docs/conceptual/runtime/contracts/contracts_module.md

```diff
@@ -0,0 +1,125 @@
+
+### Gas
+
+Contract calls are charged a gas fee to limit the amount of computational resources a transaction
+can use. When forming a contract transaction, a gas limit is specified. As the contract executes,
+gas is incrementally used up depending on the complexity of the computation. If the gas limit is
+reached before the contract execution completes, the transaction fails, contract storage is
+reverted, and the gas fee is **not** returned to the user. If the contract execution completes with
+remaining gas, it is returned to the user at the end of the transaction.
+
+The Contracts module determines the gas price, which is a conversion between the Substrate
+`Currency` and a single unit of gas. Thus, to execute a transaction, a user must have a free balance
+of at least `gas price` * `gas limit` which can be spent.
+
+### Storage Rent
+
+Similar to how gas limits the amount of computational resources that can be used during a
+transaction, storage rent limits the footprint that a contract can have on the blockchain's storage.
+A contract account is charged proportionally to the amount of storage its account uses. When a
+contract's balance goes below a defined limit, the contract's account is turned into a "tombstone"
+and its storage is cleaned up. A tombstone contract can be restored by providing the data that was
+cleaned up when it became a tombstone as well as any additional funds needed to keep the contract
+alive.
+
+## Contracts Module vs EVM
+
+The SRML Contracts module iterates on existing ideas in the smart contract ecosystem, particularly
+Ethereum and the EVM.
+
+The most obvious difference between the Contracts module and the EVM is the underlying execution
+engine used to run smart contracts. The EVM is a good theoretical execution environment, but it is
+not very practical use with modern hardware. For example, manipulation of 256 bit integers on modern
+architectures is significantly more complex than standard types. Even the Ethereum team has
+investigated the use of [Wasm](https://github.com/ewasm/design) for the next generation of the
+network.
+
+The EVM charges for storage fees only at the time of storage. This one-time cost results in some
+permanent amount of storage being used on the blockchain, _forever_, which is economically unsound.
+The SRML Contracts module attempts to repair this through [storage rent](#storage-rent) which
+ensures that any data that persists on the blockchain is appropriately charged for those resources.
+
+The Contracts module chooses to approach contract creation using a [two-step
+process](#two-step-deployment), which fundamentally changes how contracts are stored on chain.
+Contract addresses, their storage, and balances are now separated from the underlying contract
+logic. This could enable behavior like what [`create2`](https://eips.ethereum.org/EIPS/eip-1014)
+provided to Ethereum or even enable repairable or upgradeable contracts on a Substrate based
+blockchain.
```

Comment:
ehh, if there was more meat to the individual points I would, but I think because I am losely referencing differences here, they are not quite so categorized. I would also say this list is probably not the definitive set of differences here.